Sources/Basics/Concurrency/ConcurrencyHelpers.swift

```diff
@@ -27,6 +27,12 @@ public enum Concurrency {
 
 @available(*, noasync, message: "This method blocks the current thread indefinitely. Calling it from the concurrency pool can cause deadlocks")
 public func unsafe_await<T: Sendable>(_ body: @Sendable @escaping () async -> T) -> T {
+    withUnsafeCurrentTask { task in
+        if task != nil {
+            fatalError("This function must not be invoked from the Swift Concurrency thread pool.")
```

Comment:
question: will the fatal error contains the function reference?  if not, consider replading `function` with `\(#function)` and maybe even include `\(#file)` somewhere in the message .